### PR TITLE
TST: destroy databroker on finish

### PR DESCRIPTION
**Pull request by CJ-Wright**
(for those of us with small memory/paranoia)

### CODE REVIEW

**Comment by tacaswell** [comment]
I don't see how this is different.

**Comment by CJ-Wright** [comment]
I don't think that currently there is a cleanup step on the databroker so it could persist on disk

**Comment by danielballan** [comment]
I would like to make it possible to run the bluesky tests without running a mongo server, and this would be a step backward in that respect.

**Comment by tacaswell** [comment]
Ah, that makes sense.  Can you add a finalizer instead that removes the files?  I agree with @danielballan running these tests without mongo is a major :+1:

**Comment by CJ-Wright** [comment]
Ok, will get in tomorrow.

**Comment by CJ-Wright** [comment]
@tacaswell can you please assign me?

**Comment by CJ-Wright** [comment]
@danielballan we could do a try accept which checks if we can open up a mongo server, and if not falls back on sqlite.

**Comment by danielballan** [comment]
@CJ-Wright That sounds like a recipe for confusion and complexity. We can do all the bluesky testing we need to do against sqlite. Many of the databroker tests use a real RunEngine and test it against all the backends. There is no need for bluesky to test that integration as well.

File: bluesky/tests/conftest.py [hunk 1]
```diff
@@ -36,5 +37,6 @@ def motor_det(request):
 def db(request):
     """Return a data broker
     """
-    from databroker import temp_config, Broker
-    return Broker.from_config(temp_config())
+    from databroker.tests.utils import build_pymongo_backed_broker
+    db = build_pymongo_backed_broker(request)
```
Comment (tacaswell) [inline bluesky/tests/conftest.py]: Another option is to switch this to the sqlite backend factory which already has the clean up attached to it.

Comment (CJ-Wright) [inline bluesky/tests/conftest.py]: That is my plan now that people have asked for sqlite

**Comment by tacaswell** [comment]
Tests will be fixed by https://github.com/NSLS-II/databroker/pull/278

**Comment by tacaswell** [comment]
@CJ-Wright can I force-push to this branch?

**Comment by CJ-Wright** [comment]
@tacaswell of course

**Comment by CJ-Wright** [comment]
cycle for ci

**Comment by tacaswell** [comment]
It looks like the release version of databroker has a missing ujson dep.  We should just hold off on merging this until we tag the next databroker.

**Comment by CJ-Wright** [comment]
Are you going to tag databroker and then bluesky? (Is this getting in this cycle?)

**Comment by CJ-Wright** [comment]
Has there been any progress on this? The lack of test file removal almost crashed my machine.

**Comment by danielballan** [comment]
For simplicit, can `build_sqlite_backed_broker` use the new `Broker.from_config(temp_config())`? It can of course also include cleanup.

**Comment by CJ-Wright** [comment]
Sure works for me

**Comment by danielballan** [comment]
@tacaswell?

**Comment by danielballan** [comment]
See https://github.com/NSLS-II/databroker/pull/294

**Pull request merged**